src/afl-fuzz-run.c

```diff
@@ -860,26 +889,30 @@ void sync_fuzzers(afl_state_t *afl) {
 
         if (mem == MAP_FAILED) { PFATAL("Unable to mmap '%s'", path); }
 
-        /* See what happens. We rely on save_if_interesting() to catch major
-           errors and save the test case. */
+        if (!is_known_case(afl, namelist[o]->d_name, mem, st.st_size)) {
```

Comment:
if you use a simplified check as I remarked, you cna move the check upwards into the `if (st.st_size && st.st_size <= MAX_FILE) {` check which then also avoids the mmap.